test/integration/scheduler_perf/config/performance-config.yaml

```diff
@@ -952,3 +952,63 @@
       initPods: 1000
       measurePods: 1000
       maxClaimsPerNode: 10
+
+# To ensure gated Pods don't impact the throughput of the scheduler negatively.
+# https://github.com/kubernetes/kubernetes/issues/124384
+- name: SchedulingGatedPodsImpactForThroughput
+  defaultPodTemplatePath: config/templates/pod-with-label.yaml
+  workloadTemplate:
+  - opcode: createNodes
+    count: 1
+    nodeTemplatePath: config/templates/node-with-name.yaml
+  - opcode: createPods
+    countParam: $gatedPods
+    podTemplatePath: config/templates/gated-pod.yaml
+    skipWaitToCompletion: true
+  - opcode: sleep
+    # To produce a stable scheduler_perf result, here we make sure all gated Pods are stored in the scheduling queue.
+    duration: 5s
+  - opcode: churn
+    # Keep creating nodes, which result in many cluster events (Node/Add)
+    # and each of them will be processed by the scheduling queue.
+    # But, the scheduling throughput should not be impacted by the number of gated Pods.
+    mode: create
+    number: 10000
+    intervalMilliseconds: 10
+    templatePaths:
+    - config/templates/node-default.yaml
+  - opcode: createPods
+    countParam: $measurePods
+    collectMetrics: true
+  workloads:
+  - name: 1000Node
+    labels: [performance, fast]
+    params:
+      gatedPods: 10000
+      measurePods: 20000
+
+- name: SchedulingGatedPodsWithPodAffinityImpactForThroughput
```

Comment:
This test case needs to be added in any case as it's not overlapped with https://github.com/kubernetes/kubernetes/pull/125504.